versioned/storage/jdbc/src/main/java/org/projectnessie/versioned/storage/jdbc/AbstractJdbcPersist.java

```diff
@@ -515,8 +516,7 @@ private boolean[] upsertObjs(
         ps.setString(storeObjSqlParams.get(COL_REPO_ID), config.repositoryId());
         serializeObjId(ps, storeObjSqlParams.get(COL_OBJ_ID), id, databaseSpecific);
         ps.setString(storeObjSqlParams.get(COL_OBJ_TYPE), type.name());
-        String versionToken =
-            (obj instanceof UpdateableObj) ? ((UpdateableObj) obj).versionToken() : null;
+        String versionToken = UpdateableObjs.extractVersionToken(obj).orElse(null);
```

Comment:
```suggestion
        Optional<String> versionToken = UpdateableObjs.extractVersionToken(obj).orElse(null);
        if (versionToken.isPresent()) {
          ps.setString(storeObjSqlParams.get(COL_OBJ_VERS), versionToken.get());
```

(GH doesn't let me add a multi-line suggestion in this case)